front/src/applications/editor/tools/rangeEdition/components/GroupedTrackRangeList.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useContext } from 'react';
+
+import { groupBy } from 'lodash';
+
+import EditorContext from 'applications/editor/context';
+import type {
+  ApplicableTrackRange,
+  ElectrificationEntity,
+  RangeEditionState,
+  SpeedSectionEntity,
+} from 'applications/editor/tools/rangeEdition/types';
+import type { ExtendedEditorContextType } from 'applications/editor/types';
+import { partialIsEqual } from 'utils/object';
+
+import TrackRange from './TrackRange';
+
+type GroupedTrackRangeListProps = {
+  displayedRanges: ApplicableTrackRange[];
+};
+
+const GroupedTrackRangesList = ({ displayedRanges }: GroupedTrackRangeListProps) => {
+  const {
+    state: { routeElements, routeExtra = {} },
+  } = useContext(EditorContext) as ExtendedEditorContextType<
+    RangeEditionState<SpeedSectionEntity | ElectrificationEntity>
+  >;
+
+  const rangesByRoute = groupBy(displayedRanges, (currentRange) => {
+    const isCurrentRange = partialIsEqual(['track', 'begin', 'end'], currentRange);
+    // eslint-disable-next-line no-restricted-syntax
+    for (const [routeKey, { trackRanges }] of Object.entries(routeElements)) {
+      if (isCurrentRange(routeExtra[routeKey]) || trackRanges.some(isCurrentRange)) {
+        return routeKey;
+      }
+    }
+    return undefined;
+  });
+  return Object.entries(rangesByRoute).reduce<JSX.Element[]>((acc, [route, trackRanges], index) => {
+    const routeJSX = <h3 key={`route-${index}`}>{route}</h3>;
+    const routeTrackRanges = trackRanges.map((range, rangeIndex) => (
+      <TrackRange
+        range={range}
+        index={rangeIndex}
+        speedRestrictionTool
+        key={`track-range-${range.track}-${range.begin}-${range.end}-${range.applicable_directions}-${rangeIndex}`}
+      />
+    ));
+    acc.push(
+      <>
+        {routeJSX}
+        {routeTrackRanges}
```

Comment:
Could we leave the h3 here ? I don't think using a const is necessary

```suggestion
        <h3 key={`route-${index}`}>{route}</h3>
        {routeTrackRanges}
```

Reply:
Sure